Makefile

```diff
@@ -1,28 +1,67 @@
-PROJECT_ROOT := src/
-VERSION = 3.2.0
+# Makefile used to create packages for mixer-tools. It doesn't assume
+# that the code is inside a GOPATH, and always copy the files into a
+# new workspace to get the work done. Go tools doesn't reliably work
+# with symbolic links.
+#
+# For historical purposes, it also works in a development environment
+# when the repository is already inside a GOPATH.
+
+GO_PACKAGE_PREFIX := github.com/clearlinux/mixer-tools
+
+.PHONY: gopath
+
+# Strictly speaking we should check if it the directory is inside an
+# actual GOPATH, but the directory structure matching is likely enough.
+ifeq (,$(findstring ${GO_PACKAGE_PREFIX},${PWD}))
+LOCAL_GOPATH := ${PWD}/.gopath
+export GOPATH := ${LOCAL_GOPATH}
+gopath:
+	@rm -rf ${LOCAL_GOPATH}/src
+	@mkdir -p ${LOCAL_GOPATH}/src/${GO_PACKAGE_PREFIX}
+	@cp -af * ${LOCAL_GOPATH}/src/${GO_PACKAGE_PREFIX}
+	@echo "Prepared a local GOPATH=${GOPATH}"
+else
+LOCAL_GOPATH :=
+GOPATH ?= ${HOME}/go
+gopath:
+	@echo "Code already in existing GOPATH=${GOPATH}"
+endif
+
+.PHONY: all compliant build install clean
 
 .DEFAULT_GOAL := all
+all: gopath compliant build
```

Comment:
I would think about removing gopath from all.

I am not sure how good an idea it is to be potentially running compliant and build in parallel, if compliant updates a file as build is reading it. Probably will not happen in practice as the code will be formatted almost all the time.

Reply:
Thanks for noticing this. Added .NOTPARALLEL to the whole Makefile.